src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -941,11 +941,77 @@ protected function send_plugin_theme_email( $type, $successful_updates, $failed_
 			return;
 		}
 
+		$unique_failures = false;
+		$failure_emails  = get_option( 'failed_plugin_theme_update_emails', array() );
+
+		// When only failures have occurred, an email should only be sent if there are either new failures,
+		// or failures that have not sent out an email recently.
+		if ( 'fail' === $type ) {
+			$unique_failures = false;
+
+			/**
+			 *
+			 */
+			$failure_email_interval = apply_filters( 'auto_plugin_theme_update_fail_email_interval', DAY_IN_SECONDS * 3 );
+
+			foreach ( $failed_updates as $update_type => $failures ) {
+				foreach ( $failures as $failed_update ) {
+					if ( ! isset( $failure_emails[ $failed_update->item->{$update_type} ] ) ) {
+						$unique_failures = true;
+						continue;
+					}
+
+					// Enough time has passed to make this failure warrant an email.
+					if ( time() - $failure_emails[ $failed_update->item->{$update_type} ] > $failure_email_interval ) {
+						$unique_failures = true;
+					}
+				}
+			}
+
+			if ( ! $unique_failures ) {
+				return;
+			}
+		}
+
 		$body               = array();
 		$successful_plugins = ( ! empty( $successful_updates['plugin'] ) );
 		$successful_themes  = ( ! empty( $successful_updates['theme'] ) );
 		$failed_plugins     = ( ! empty( $failed_updates['plugin'] ) );
 		$failed_themes      = ( ! empty( $failed_updates['theme'] ) );
+		
+		// remove from the failed_plugin_theme_update_emails list if plugin or theme version was update manually or removed by admin
+		if( $successful_plugins || $successful_themes || $failed_plugins || $failed_themes )
+		{
+			$temp_merge_all = array();
+			
+			if( $successful_plugins ) {
+				foreach ( $successful_updates['plugin'] as $item ) {
+					// set key for the intersect
+					$temp_merge_all[ $item->item->plugin ] = $item->name; 
+				}
+			}
+			if( $successful_themes ) {
+				foreach ( $successful_updates['theme'] as $item ) {
+					// set key for the intersect
+					$temp_merge_all[ $item->item->theme ] = $item->name; 
+				}
+			}
+			if( $failed_plugins ) {
+				foreach ( $failed_updates['plugin'] as $item ) {
+					// set key for the intersect
+					$temp_merge_all[ $item->item->plugin ] = $item->name;  
+				}
+			}
+			if( $failed_themes ) {
+				foreach ( $failed_updates['theme'] as $item ) {
+					// set key for the intersect
+					$temp_merge_all[ $item->item->theme ] = $item->name; 
+				}
+			}
+			
+			$failure_emails = array_intersect_key( $failure_emails, $temp_merge_all );
+		}
+		
```

Comment:
This code is going check with whenever the user goes manually applies an update, whether through clicking update in the dashboard or by uploading a new version of the plugin or theme manually.

Reply:
This won't have the intended affect. This function is not run during an update initiated by clicking update in the admin. Incoming commit with a different approach.